src/demo-app/dialog/dialog-demo.html

```diff
@@ -1,53 +1,107 @@
-      </md-input-container>
-      <md-input-container>
-        <input mdInput [(ngModel)]="config.height" placeholder="Height">
-      </md-input-container>
-    </p>
-
-    <h2>Dialog position</h2>
-
-    <p>
-      <md-input-container>
-        <input mdInput [(ngModel)]="config.position.top" (change)="config.position.bottom = ''" placeholder="Top">
-      </md-input-container>
-      <md-input-container>
-        <input mdInput [(ngModel)]="config.position.bottom" (change)="config.position.top = ''" placeholder="Bottom">
-      </md-input-container>
-    </p>
-
-    <p>
-      <md-input-container>
-        <input mdInput [(ngModel)]="config.position.left" (change)="config.position.right = ''" placeholder="Left">
-      </md-input-container>
-      <md-input-container>
-        <input mdInput [(ngModel)]="config.position.right" (change)="config.position.left = ''" placeholder="Right">
-      </md-input-container>
-    </p>
-
-    <h2>Other options</h2>
-
-    <p>
-      <md-select placeholder="Button alignment" [(ngModel)]="actionsAlignment">
-        <md-option>Start</md-option>
-        <md-option value="end">End</md-option>
-        <md-option value="center">Center</md-option>
-      </md-select>
-    </p>
-
-    <md-checkbox [(ngModel)]="config.disableClose">Disable close</md-checkbox>
-  </md-card-content>
-</md-card>
-
-<p>Last close result: {{lastCloseResult}}</p>
+<div class="demo-dialog-section">
+
+  <h2>Using <strong>MdDialog</strong> service</h2>
+
+  <div class="demo-button-group">
+    <button md-raised-button color="primary" (click)="openJazz()" [disabled]="dialogRef">Open dialog</button>
+    <button md-raised-button color="accent" (click)="openContentElement()">Open dialog with content elements</button>
+  </div>
+
+  <div class="demo-button-group">
+    <button md-raised-button color="primary" (click)="openJazzUsingTemplateRef()" [disabled]="dialogTemplateRef">Open dialog using TemplateRef</button>
+    <button md-raised-button color="accent" (click)="openContentElementUsingTemplateRef()">Open dialog using TemplateRef with content elements</button>
+  </div>
+
+  <p *ngIf="lastCloseResult">Last close result: {{lastCloseResult}}</p>
+
+  <template #jazzDialogRef>
+    <demo-jazz-dialog-template-ref (close)="closeJazzUsingTemplateRef($event)"></demo-jazz-dialog-template-ref>
+  </template>
+
+  <template #contentElementRef>
+    <demo-content-element-template-ref-dialog (close)="closeContentElementUsingTemplateRef()" [actionsAlignment]="actionsAlignment"></demo-content-element-template-ref-dialog>
+  </template>
+
+</div>
+
+<div class="demo-dialog-section">
+
+  <h2>Using <strong>md-dialog</strong> element</h2>
+
+  <div class="demo-button-group">
+    <button
+      md-raised-button
+      color="primary"
+      (click)="changeMdDialogState(true)"
+      [disabled]="mdDialogState">
+      Open md-dialog with content elements
+    </button>
+  </div>
+
+  <md-dialog
```

Comment:
Not a full review for now, but I would do this as a directive that is meant to go on a template element, so it would be either of
```html
<template mdDialog ...>

or

<div *mdDialog ...>
```

Mainly so that it doesn't leave a phantom `md-dialog` element in the DOM